cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureVirtualMachineService.java

```diff
@@ -51,7 +51,19 @@
         while (hasMissingVm(virtualMachines, privateInstanceIds) && virtualMachines.hasNextPage()) {
             virtualMachines.loadNextPage();
         }
+        if (hasNoVmInResourceGroup(virtualMachines)) {
+            LOGGER.info("We could not receive any VM in resource group. Let's try to fetch VMs by instance ids.");
+            for (String privateInstanceId : privateInstanceIds) {
+                VirtualMachine virtualMachineByResourceGroup = azureClient.getVirtualMachineByResourceGroup(resourceGroup, privateInstanceId);
```

Comment:
Wonder how would azure react if we would tried to find many instances, wouldn't we reach a ratelimit problem. 
We are just verifying that instances were created, can we limit the max. number of calls here in the expense of some check are skipped in case of high nodecount clusters?